contracts/orderValidation/OrderValidatorV1.sol

```diff
@@ -0,0 +1,439 @@
+        }
+
+        if (!isApprovedAll && !isApprovedSingle) return ERC721_NO_APPROVAL_FOR_ALL_OR_TOKEN_ID;
+        if ((IERC721(collection).ownerOf(tokenId)) != user) return ERC721_TOKEN_ID_NOT_IN_BALANCE;
+    }
+
+    /**
+     * @notice Check validity of ERC1155 approvals and balances required to process the order
+     * @param collection address of the collection
+     * @param user address of the user
+     * @param transferManager address of the transfer manager
+     * @param tokenId tokenId
+     * @param amount amount
+     */
+    function _validateERC1155(
+        address collection,
+        address user,
+        address transferManager,
+        uint256 tokenId,
+        uint256 amount
+    ) internal view returns (uint256 validationCode) {
+        bool isApprovedAll = IERC1155(collection).isApprovedForAll(user, transferManager);
+        if (!isApprovedAll) return ERC1155_NO_APPROVAL_FOR_ALL;
+        if ((IERC1155(collection).balanceOf(user, tokenId)) < amount)
+            return ERC1155_BALANCE_TOKEN_ID_INFERIOR_TO_AMOUNT;
+    }
+
+    /**
+     * @notice Check validity of EOA maker order
+     * @param digest digest
+     * @param targetSigner the signer address to confirm message validity
+     * @param v parameter (27 or 28). This prevents maleability since the public key recovery equation has two possible solutions.
+     * @param r parameter
+     * @param s parameter
+     */
+    function _validateEOA(
+        bytes32 digest,
+        address targetSigner,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) internal pure returns (uint256 validationCode) {
+        if (uint256(s) > 0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF5D576E7357A4501DDFE92F46681B20A0)
+            return INVALID_S_PARAMETER_EOA;
+
+        if (v != 27 && v != 28) return INVALID_V_PARAMETER_EOA;
+
+        address signer = ecrecover(digest, v, r, s);
+        if (signer == address(0)) return NULL_SIGNER_EOA;
+        if (signer != targetSigner) return WRONG_SIGNER_EOA;
+    }
+
+    /**
+     * @notice Check validity of ERC-1271 maker order
+     * @param digest digest
+     * @param targetSigner the signer address to confirm message validity
+     * @param v parameter (27 or 28)
+     * @param r parameter
+     * @param s parameter
+     */
+    function _validateERC1271(
+        bytes32 digest,
+        address targetSigner,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) internal view returns (uint256 validationCode) {
+        (bool answer, bytes memory data) = targetSigner.staticcall(
+            abi.encodeWithSelector(IERC1271.isValidSignature.selector, digest, abi.encodePacked(r, s, v))
+        );
+
+        if (!answer) return MISSING_IS_VALID_SIGNATURE_FUNCTION_EIP1271;
+        bytes4 magicValue = abi.decode(data, (bytes4));
+
+        // 0x1626ba7e is the interfaceId for signature contracts (see IERC1271)
+        if (magicValue != 0x1626ba7e) return SIGNATURE_INVALID_EIP1271;
```

Comment:
Move `0x1626ba7e` to a constant to be consistent with how we handle other interface IDs?

Reply:
Done